src/main/java/executor/service/service/parallel/TaskWorker.java

```diff
@@ -0,0 +1,46 @@
+package executor.service.service.parallel;
+
+import executor.service.service.ItemHandler;
+import executor.service.service.ProxySourcesClient;
+import executor.service.service.ScenarioSourceListener;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.function.Consumer;
+
+/**
+ * Worker tasks.
+ *
+ * @author Oleksandr Tuleninov, Yurii Kotsiuba.
+ * @version 01
+ * */
+public class TaskWorker<T> implements Runnable {
+
+    private final T listener;
+    private final BlockingQueue<T> queue;
+
+    public TaskWorker(T listener, BlockingQueue<T> queue) {
+        this.listener = listener;
+        this.queue = queue;
+    }
+
+    @Override
+    public void run() {
+        Consumer<T> itemHandlerConsumer = items -> {
+            try {
+                queue.put(items);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+        };
+
+        if (listener instanceof ScenarioSourceListener) {
+            ((ScenarioSourceListener) listener).execute(createItemHandler(itemHandlerConsumer));
+        } else if (listener instanceof ProxySourcesClient) {
+            ((ProxySourcesClient) listener).execute(createItemHandler(itemHandlerConsumer));
+        }
+    }
```

Comment:
In this specific implementation, using generics is unnecessary because it restricts the allowed types to just those two classes. What is purpose of this generic class when in fact it can only handle either Scenarios or Proxies?  Generics are typically used to make code more flexible by allowing it to work with various types. When they are used in a way that appears to restrict the allowed types, it can lead to confusion and misunderstanding.